ibc/src/component/channel.rs

```diff
@@ -0,0 +1,269 @@
+use crate::component::client::View as _;
+use crate::component::connection::View as _;
+use anyhow::Result;
+use async_trait::async_trait;
+use ibc::core::ics02_client::client_consensus::ConsensusState;
+use ibc::core::ics02_client::client_def::AnyClient;
+use ibc::core::ics02_client::client_def::ClientDef;
+use ibc::core::ics02_client::client_state::ClientState;
+use ibc::core::ics03_connection::connection::{ConnectionEnd, State as ConnectionState};
+use ibc::core::ics04_channel::channel::State as ChannelState;
+use ibc::core::ics04_channel::channel::{ChannelEnd, Counterparty};
+use ibc::core::ics04_channel::msgs::chan_close_confirm::MsgChannelCloseConfirm;
+use ibc::core::ics04_channel::msgs::chan_close_init::MsgChannelCloseInit;
+use ibc::core::ics04_channel::msgs::chan_open_ack::MsgChannelOpenAck;
+use ibc::core::ics04_channel::msgs::chan_open_confirm::MsgChannelOpenConfirm;
+use ibc::core::ics04_channel::msgs::chan_open_init::MsgChannelOpenInit;
+use ibc::core::ics04_channel::msgs::chan_open_try::MsgChannelOpenTry;
+use ibc::core::ics24_host::identifier::ChannelId;
+use ibc::core::ics24_host::identifier::PortId;
+use penumbra_chain::genesis;
+use penumbra_component::Component;
+use penumbra_proto::ibc::ibc_action::Action::{
+    ChannelCloseConfirm, ChannelCloseInit, ChannelOpenAck, ChannelOpenConfirm, ChannelOpenInit,
+    ChannelOpenTry,
+};
+use penumbra_storage::{State, StateExt};
+use penumbra_transaction::Transaction;
+use tendermint::abci;
+use tracing::instrument;
+
+mod stateful;
+mod stateless;
+
+mod execution;
+
+pub struct ICS4Channel {
```

Comment:
nit: could we rename this to `Ics4Channel` for consistency with the other IBC components?